### PR TITLE
Render pagination links when the destinations are empty strings

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Render pagination links when the destinations are empty strings.

## Why?

AFAICS this is causing a problem because of a change upstream in frontend (https://github.com/guardian/frontend/pull/28825). https://github.com/guardian/frontend/pull/28825/changes/12355aa4c66294632353139fd0b110082ee03ebf removed a query param which was populating pagination strings:

| Before      | After      |
| ----------- | ---------- |
| Screenshot 2026-06-14 at 21 09 53 | Screenshot 2026-06-14 at 21 10 05 |

My understanding is that this is reasonable, as the string is intended to be a query string, and an empty query string in this case just means 'latest', i.e. there's no preceding block to mention.

Downstream in DCR, the check to see whether pagination strings are present is ... you guessed it:

https://github.com/guardian/dotcom-rendering/blob/648a9d809939a5d29bad7681ae4d6f1b2084272f/dotcom-rendering/src/components/Pagination.tsx#L96

... which will ignore empty strings. (The linter currently warns, but does not error, on this line.)

This PR makes this check an explicit comparison to `undefined`, and adds unit tests. The tests were added prior to the fix to verify red/green. If this'd be better verified with a story, super happy to add!

~Finally, do we need the tests for Next/Oldest? I can't imagine a scenario where they'd be empty strings (they have to reference a preceding block to be paginated, no?) Perhaps replacing the empty strings in those cases would be more representative.~ I've done that in [579cda9](https://github.com/guardian/dotcom-rendering/pull/16153/commits/579cda92fd68bb6767207b3ec62be0e848bdf0d4).

NB: The investigation and the fix was manual. The tests were written with Claude Opus 4.8, and reviewed manually.

## Screenshots

From [this piece](https://www.theguardian.com/australia-news/live/2026/jun/14/australia-news-live-jonathon-duniam-senator-retire-liberal-national-coalition-labor-albanese-taylor-one-nation-sydney-nsw-victoria-high-court-ntwnfb?page=with%3Ablock-6a2e04868f0881a06771523f#liveblog-navigation), local DCAR pointing at CODE CAPI:

| Before      | After      |
| ----------- | ---------- |
| Screenshot 2026-06-14 at 21 30 54 | Screenshot 2026-06-14 at 21 30 40 |

### CODE REVIEW

File: dotcom-rendering/src/components/Pagination.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const Pagination = ({
 					style={{ gridArea: 'newer', justifySelf: 'start' }}
 					css={flexSection}
 				>
-					{!!newest && (
+					{newest !== undefined && (
 						<>
 							<Hide from="phablet">
 								<LinkButton
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply dotcom-rendering/src/components/Pagination.tsx:105]: Good bot! Addressed in [2fb5a73](https://github.com/guardian/dotcom-rendering/pull/16153/commits/2fb5a73499947d65391a798a68dd5bba8e1ea531), with test.

**Review by RikRootsGuardian — APPROVED** [review]
Happy to approve, with CoPilot suggestion added

**Review by Jakeii — COMMENTED** [review]
Thank's for sorting this one out!

File: dotcom-rendering/src/components/Pagination.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply dotcom-rendering/src/components/Pagination.tsx:105]: In retrospect ... bad bot, as this had the unintended consequence of making the links contain just a fragment, resulting in navigation not working as expected when the query string was empty. Reverted in https://github.com/guardian/dotcom-rendering/pull/16156. Poor verification on my part, so also: bad @jonathonherbert.
